resources/js/pack-it-forms.js

```diff
@@ -29,18 +30,29 @@ execution after the page loads.  This is implements the mechanism, the
 calls registering startup functions are at the end of this file. */
 var startup_functions = new Array();
 
+function alert_error(e) {
+    if (e && e.stack) {
+        alert(e.stack);
+    } else {
+        try {
+            throw(new Error(JSON.stringify(e)));
+        } catch(f) {
+            alert(f.stack);
+        }
+    }
+}
+
 function startup() {
     /* The startup functions are called in continuation-passing style,
     so that they can contain asynchronous code that moves to the next
     function when it is complete. */
-    startup_functions.shift()(callclist(startup_functions));
-
-    function callclist(functions) {
-        return function() {
-            if (functions.length > 0) {
-                functions.shift()(callclist(functions));
-            }
-        };
+    var next = (startup_functions.length > 0) && startup_functions.shift();
+    if (next) {
+        try {
+            next(startup);
+        } catch(e) {
+            alert_error(e);
+        }
```

Comment:
I've experienced several errors that simply left the page showing the 'loading' spinner, without any diagnostic information.

Reply:
I think that experience should only occur for forms creators and or/ pack-it-forms developers that are making changes to forms with errors in them, so it won't affect users. I agree that it isn't the best user experience but we haven't focused on providing a better one because so far we have almost always been able to quickly figure out what went wrong using developer tools in whichever browser has the problem.

As I said, I'm not opposed to improving error handling and the user experience here. I'd just like to do it without using alert dialog boxes but in the page instead. The thing I would prefer not to change is the CPS style implementation because it has the property that the startup function list isn't mutated.